lib/slack/web/client.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Slack.Web.Client do
+  @moduledoc """
+  Defines a custom client for making calls to Slack Web API.
+  """
+
+  @type url :: String.t()
+  @type form_body :: {:form, Keyword.t()}
+  @type multipart_form_body :: {:multipart, nonempty_list(tuple())}
+  @type body :: form_body() | multipart_form_body()
```

Comment:
@BlakeWilliams I added these from what I could infer from web.ex. I didn't plan for them to be exhaustive, but just to give the hint that we're sending form data and not JSON.